storage/conformance_test.go

```diff
@@ -16,21 +16,315 @@ package storage
+							if err := deleteRetryTest(host, testID); err != nil {
+								t.Errorf("deleting retry test: %v", err)
+							}
+
+						})
+					}
+
+				}
+			}
+		}
+	}
+
+}
+
+// Create a retry test resource in the emulator. Returns the ID to pass as
+// a header in the test execution.
+func createRetryTest(host string, instructions map[string][]string) (string, error) {
+	endpoint := host + "/retry_test"
+	c := http.DefaultClient
+	data := struct {
+		Instructions map[string][]string `json:"test_instructions"`
+	}{
+		Instructions: instructions,
+	}
+
+	buf := new(bytes.Buffer)
+	if err := json.NewEncoder(buf).Encode(data); err != nil {
+		return "", fmt.Errorf("encoding request: %v", err)
+	}
+	resp, err := c.Post(endpoint, "application/json", buf)
+	if err != nil || resp.StatusCode != 200 {
+		return "", fmt.Errorf("creating retry test: err: %v, resp: %+v", err, resp)
+	}
+	defer resp.Body.Close()
+	testRes := struct {
+		TestID string `json:"id"`
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return "", fmt.Errorf("decoding test ID: %v", err)
+	}
+	return testRes.TestID, nil
+}
+
+// Verify that all instructions for a given retry testID have been used up.
+func checkRetryTest(host, testID string) error {
+	endpoint := strings.Join([]string{host, "retry_test", testID}, "/")
+	c := http.DefaultClient
+	resp, err := c.Get(endpoint)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("getting retry test: err: %v, resp: %+v", err, resp)
+	}
+	defer resp.Body.Close()
+	testRes := struct {
+		Instructions map[string][]string
+		Completed bool
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return fmt.Errorf("decoding response: %v", err)
+	}
+	if !testRes.Completed {
+		return fmt.Errorf("test not completed; unused instructions: %+v", testRes.Instructions)
+	}
+	return nil
+}
+
+// Delete a retry test resource.
+func deleteRetryTest(host, testID string) error {
+	endpoint := strings.Join([]string{host, "retry_test", testID}, "/")
+	c := http.DefaultClient
+	req, err := http.NewRequest("DELETE", endpoint, nil)
+	if err != nil {
+		return fmt.Errorf("creating request: %v", err)
+	}
+	resp, err := c.Do(req)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("deleting test: err: %v, resp: %+v", err, resp)
+	}
+	return nil
+}
+
+type withTestID struct {
+	rt     http.RoundTripper
+	testID string
+}
+
+func (wt *withTestID) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Add("x-retry-test-id", wt.testID)
+	resp, err := wt.rt.RoundTrip(r)
+	//if err != nil{
+	//	log.Printf("Error: %+v", err)
+	//}
+	return resp, err
+}
+
+// Create custom client that sends instruction
+func wrappedClient(host, testID string) (*Client, error) {
+	ctx := context.Background()
+	base := http.DefaultTransport
+	trans, err := htransport.NewTransport(ctx, base, option.WithScopes(raw.DevstorageFullControlScope),
+		option.WithUserAgent("custom-user-agent"))
```

Comment:
You don't  need to supply `option`'s do you?

Reply:
Yeah that's true, it's only needed for the real service I believe. Will remove.